Renamed packages and namespaces

colombod opened the pull request:
This pr is 
*  adopting pi-top as root name for nuget packages
*  renaming PiTopMakerArchitecture.* to PtTop.MakerArchitecture.*
*  adding first part of expansionPlate work

colombod committed:
create stating point for mmk

colombod committed:
mmk servos and motors are all using i2c

colombod committed:
adding const for i2c bus

colombod committed:
ues device

colombod committed:
calculate orientation and heading

colombod committed:
formatting

colombod committed:
check if conversion is required

colombod committed:
refactor unit creation

colombod committed:
refactoring and use foundation plate via composition, not inheritance

colombod committed:
renamed PiTopMakerArchitecture to PiTop.MakerArchitecture

colombod committed:
clean up

colombod committed:
rename the packages

colombod committed:
update appveyor

colombod committed:
adding skeleton for rover robot

colombod committed:
change signature for extensions classes

colombod committed:
add state

colombod committed:
add file

paulmey committed:
busses -> devices

paulmey committed:
Add SMBusDevice

paulmey committed:
MCU is a SMBus device

paulmey committed:
Motor example

colombod committed:
uh one tests pass!

colombod committed:
implement encoder motor

colombod committed:
measures

colombod committed:
progress

colombod committed:
removed direction enum

colombod committed:
fix servo as proper class

paulmey committed:
Add signed short interface for SMBusDevice

colombod committed:
add registers to servo

colombod committed:
initial part

colombod committed:
math support and tests

paulmey committed:
Add Servo's, fix 32 bit reads/writes

colombod committed:
sample app for rover robot

paulmey committed:
Add joystick control

m-roberts commented on src/PiTop.nuget/PiTop.nuget.csproj:
```diff
@@ -5,17 +5,17 @@
+    <PackageId>pi-top</PackageId>
     <IsPackable>true</IsPackable>
-    <PackageDescription>piTop module.</PackageDescription>
+    <PackageDescription>pi-top board.</PackageDescription>
```
The device is a `pi-top [4]`, and it is not a board. Let's try and come up with a better description for this! 😄

colombod replied:
Please let us know as we are almost ready for merge:)

colombod committed:
Merge remote-tracking branch 'origin/xbox-control' into mmk

colombod committed:
add joystick to project

paulmey committed:
wip

colombod committed:
changes to simplify 


exntesnions

colombod committed:
corrected tilt

colombod committed:
some cleanup

colombod committed:
rover robot fixes

colombod committed:
blink

colombod committed:
the lights are on

colombod committed:
improvement

colombod committed:
add configuration

colombod committed:
update package dependencies and nuget config sources

colombod committed:
fix naming

colombod committed:
pocket logger

colombod committed:
logs

colombod committed:
more logging

paulmey committed:
New prototype

colombod committed:
rewiring for log done

paulmey committed:
Add pin number to IOException when opening pin

paulmey committed:
Add multi-segment interpolate and deadzone

paulmey committed:
Add comment on EncoderMotor

colombod committed:
change log operations

paulmey committed:
Remove old project file

paulmey committed:
Add mjpg-streamer camera

paulmey committed:
Add better exception to XboxController ctor

paulmey committed:
Make camera optional

paulmey committed:
Update prototype

colombod committed:
paulified

colombod committed:
fix

colombod committed:
onnx

paulmey committed:
Fix bug in ultrasonic sensor delay

paulmey committed:
Instrument ultrasonic sensor

paulmey committed:
Study ultrasonic sensor peformance

paulmey committed:
Use WaitForEvent in UltraSonicSensor

paulmey committed:
Enable UltraSonicSensor logs

colombod committed:
format code

colombod committed:
log formatting

colombod committed:
add new package to build

colombod committed:
add filters

colombod committed:
add math dependencies

colombod committed:
clamp time

colombod committed:
clamp ticks

colombod committed:
10 seconds timeout on read

colombod committed:
better expection

colombod committed:
add more logging

colombod committed:
more logging

colombod committed:
expression

colombod committed:
giving up on timeouts

colombod committed:
update nuget packages like there is no tomorrow

colombod committed:
comment

paulmey committed:
wip

colombod committed:
filter configuration

paulmey committed:
Fix race in ultrasound wait loop

Unhandled exception. System.ArgumentOutOfRangeException: Number must be either non-negative and less than or equal to Int32.MaxValue or -1. (Parameter 'timeout')
   at System.Threading.WaitHandle.ToTimeoutMilliseconds(TimeSpan timeout)
   at System.Threading.Thread.Sleep(TimeSpan timeout)
   at PiTop.MakerArchitecture.Foundation.Sensors.UltrasonicSensor.GetDistance() in C:\src\pi-top-4-.NET-Core-API\src\PiTop.MakerArchitecture.Foundation\Sensors\UltrasonicSensor.cs:line 62

paulmey committed:
Split camera so that it can rely on system-managed mjpg_streamer

m-roberts commented on src/PiTop.Camera/StreamingServer.cs:
```diff
@@ -0,0 +1,26 @@
+﻿using System;
```
This file should be deleted

colombod replied:
deleted

m-roberts commented on src/PiTop.MakerArchitecture.Expansion/MathHelpers.cs:
```diff
@@ -0,0 +1,37 @@
+﻿using System.Linq;
```
These functions should move into the Algorithms section

colombod replied:
moved and moved tests too

colombod replied:
done

m-roberts commented:
Move rover.md closer to rover sample.
Review `libs/` - can we move them to NuGet? Or at least standardise the files in there.. include version numbers, source, build scripts, etc? Let's add 32 and 64 bit versions of each.
Need to open an issue with the .NET folks re: SMBus device - can we move this out of this repo?

m-roberts reviewed (changes requested):
@colombod 😎

m-roberts commented:
Create NuGet packages for native runtimes

colombod committed:
dashboard prototype


Pr feedback


rename project folder


moved all sample into folder src/Examples


add files for runtime nuget builds


increase resolution


renamed platform for tunrime


enable more logs

colombod committed:
adding exceptions

colombod committed:
add all files to package

colombod committed:
updated pocket logger

colombod committed:
add readme

m-roberts commented on runtimes/Microsoft.Onnx.Runtime/1.5.2/pit-top.Onnx.runtime.nuspec:
```diff
@@ -0,0 +1,29 @@
+    <metadata>
+        <id>pit-top.Onnx.runtime</id>
+        <version>1.5.2</version>
+        <title>Onnx native bindings for pit-top OS</title>
```
`pit-top` -> `pi-top`

m-roberts replied:
This occurs in a couple of places 😝

colombod replied:
because I am good

colombod committed:
spelling errors
